test/hydrated_test.dart

```diff
@@ -169,40 +201,38 @@ void main() {
       });
 
       test(
-          'given persistence interface `get` throws an Exception, '
-          'constructing the HydratedSubject throws an asynchronous uncatchable error',
-          () {
-        mockKeyValueStore.getOverride = (_) async => throw Exception('test');
-        runZonedGuarded(
-          () {
-            final completer = Completer();
-            HydratedSubject<int>(
-              key,
-              keyValueStore: mockKeyValueStore,
-              onHydrate: completer.complete,
-            );
-            return completer.future;
-          },
-          expectAsync2((error, _) {
-            expect(error, isA<Exception>());
-          }, count: 1),
+          'given persistence interface get throws an Exception, '
+          'constructing the HydratedSubject propagates the exception',
+          () async {
+        mockKeyValueStore.getOverride = (_) async {
+          throw Exception('test');
+        };
+        final s = HydratedSubject<int>(
+          key,
+          keyValueStore: mockKeyValueStore,
         );
+
+        await pumpEventQueue();
+
+        expect(s.error, isA<Exception>());
       });
```

Comment:
This change comes from the fact that now we propagate all exceptions to the `HydratedSubject` Stream.

This gives user the freedom of choice when it comes to catching Errors.